core/dataloader/util.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dataloader
+
+import "io"
+
+// stringPrefixReader is an io.ReadCloser that reads from a string prefix before reading from
+// another io.Reader. This is used for reassembling partial records across multi-message
+// exchanges, since the end of a wire message does not typically line up with the end of a record.
+type stringPrefixReader struct {
```

Comment:
Seems like this file should be `string_prefix_reader` instead of `util`, but it's NBD either way

Reply:
Good call! That's better than the generic `util` name, and there's nothing else in that file anyway. Fixed in 34c41be309bee05e2c67a3cd3f93a21f96762688.